app/org/thp/cortex/util/docker/DockerClient.scala

```diff
@@ -0,0 +1,167 @@
+    if (Files.exists(jobDirectory.resolve("input").resolve("cacerts")))
+      containerCmd.withEnv(s"REQUESTS_CA_BUNDLE=/job/input/cacerts")
+    val containerResponse = containerCmd.exec()
+    logger.info(
+      s"about to start container ${containerResponse.getId}\n" +
+        s"  timeout: ${timeout.toString}\n" +
+        s"  image  : $image\n" +
+        s"  volumes : ${jobDirectory.toAbsolutePath}"
+    )
+    if (containerResponse.getWarnings.nonEmpty) logger.warn(s"${containerResponse.getWarnings.mkString(", ")}")
+    scheduleContainerTimeout(containerResponse.getId, timeout)
+
+    containerResponse.getId
+  }
+
+  private def configure(jobDirectory: Path, jobBaseDirectory: Path, dockerJobBaseDirectory: Path): HostConfig = {
+    val hostConfigMut = HostConfig
+      .newHostConfig()
+      .withBinds(
+        Seq(
+          new Bind(
+            dockerJobBaseDirectory.resolve(jobBaseDirectory.relativize(jobDirectory)).toAbsolutePath.toString,
+            new Volume(s"/job"),
+            AccessMode.rw
+          )
+        ): _*
+      )
+
+    config.getOptional[Seq[String]]("docker.container.capAdd").map(_.map(Capability.valueOf)).foreach(hostConfigMut.withCapAdd(_: _*))
+    config.getOptional[Seq[String]]("docker.container.capDrop").map(_.map(Capability.valueOf)).foreach(hostConfigMut.withCapDrop(_: _*))
+    config.getOptional[String]("docker.container.cgroupParent").foreach(hostConfigMut.withCgroupParent)
+    config.getOptional[Long]("docker.container.cpuPeriod").foreach(hostConfigMut.withCpuPeriod(_))
+    config.getOptional[Long]("docker.container.cpuQuota").foreach(hostConfigMut.withCpuQuota(_))
+    config.getOptional[Seq[String]]("docker.container.dns").map(_.asJava).foreach(hostConfigMut.withDns)
+    config.getOptional[Seq[String]]("docker.container.dnsSearch").map(_.asJava).foreach(hostConfigMut.withDnsSearch)
+    config.getOptional[Seq[String]]("docker.container.extraHosts").foreach(l => hostConfigMut.withExtraHosts(l: _*))
+    config.getOptional[Long]("docker.container.kernelMemory").foreach(hostConfigMut.withKernelMemory(_))
+    config.getOptional[Long]("docker.container.memoryReservation").foreach(hostConfigMut.withMemoryReservation(_))
+    config.getOptional[Long]("docker.container.memory").foreach(hostConfigMut.withMemory(_))
+    config.getOptional[Long]("docker.container.memorySwap").foreach(hostConfigMut.withMemorySwap(_))
+    config.getOptional[Long]("docker.container.memorySwappiness").foreach(hostConfigMut.withMemorySwappiness(_))
+    config.getOptional[String]("docker.container.networkMode").foreach(hostConfigMut.withNetworkMode)
+    config.getOptional[Boolean]("docker.container.privileged").foreach(hostConfigMut.withPrivileged(_))
+
+    hostConfigMut
+  }
+
+  def info: Info = underlyingClient.infoCmd().exec()
+  def pullImage(image: String, timeout: FiniteDuration): Boolean = {
+    val pullImageResultCbk = underlyingClient // Blocking
```

Comment:
blocking calls should be surrounded by `scala.concurrent.blocking`

Reply:
yes but it is useful only on Future context apparently `You can use it anywhere, but it will only have an effect on the worker threads, i.e. computations executed with Future { ... }`

Reply:
In PlayFramework, everything is in a future, no ?